File: php/hamming/hamming.php
<?php

function distance($a, $b)
{
  if (strlen($a) !== strlen($b)) {
      throw new InvalidArgumentException('DNA strands must be of equal length.');
  }
  $distance = 0;
  for ($i = 0; $i < strlen($a); $i++) {
      $distance += $a[$i] === $b[$i] ? 0 : 1;
  }
  return $distance;
}

File: go/hamming/hamming.go
package hamming

import "errors"

// Distance calculates the Hamming distance between dna strands.
func Distance(a, b string) (int, error) {
	distance := 0
	if len(a) != len(b) {
		return -1, errors.New("strand lengths are not equal")
	}
	for i, char := range a {
		if rune(b[i]) != char {
			distance++
		}
	}
	return distance, nil
}
